### PR TITLE
Bug 1834970 - Embed the side-by-side job links in Perfherder alerts view

### PR DESCRIPTION
**Pull request by alexandru-io**

### CODE REVIEW

File: ui/perfherder/alerts/AlertTable.jsx [hunk 1]
```diff
@@ -355,6 +360,12 @@ export default class AlertTable extends React.Component {
                         onChangeSort={this.onChangeSort}
                       />
                     </th>
+                    <th className="align-bottom">
```
Comment (esanuandra) [inline ui/perfherder/alerts/AlertTable.jsx:363]: Maybe we could add '`text-nowrap`' class.

File: ui/perfherder/perf-helpers/perfdocs.js [hunk 2]
```diff
@@ -25,87 +27,11 @@ const supportedPerfdocsFrameworks = {
...
-  'youtube-playback-widevine-h264-sfr',
-  'youtube-playback-widevine-hfr',
-  'youtube-playback-widevine-hfr',
-];
-
 /**
  * TODO: remove hardcoded names once suffixes are removed from Perfdocs
  * @link https://firefox-source-docs.mozilla.org/testing/perfdocs/raptor.html#interactive
  */
+
```
Comment (beatrice-acasandrei) [inline ui/perfherder/perf-helpers/perfdocs.js:34]:
```suggestion
```

File: ui/perfherder/alerts/AlertTableRow.jsx [hunk 3]
```diff
@@ -45,14 +51,22 @@ export default class AlertTableRow extends React.Component {
...
       starred: this.props.alert.starred,
       checkboxSelected: false,
+      showSideBySideLink: false,
       icons: [],
     };
   }
 
   componentDidMount() {
     const { alert } = this.props;
+    const showSideBySideLink =
```
Comment (beatrice-acasandrei) [inline ui/perfherder/alerts/AlertTableRow.jsx:61]: I believe this logic can be moved in `render()`. I don't see why `showSideBySideLink` should be kept in state.

File: ui/perfherder/perf-helpers/constants.js [hunk 4]
```diff
@@ -202,3 +202,107 @@ export const noiseProfiles = {
...
+  'fandom',
+  'google-slides',
+  'instagram',
+  'twitter',
+  'wikipedia',
+  'yahoo-mail',
+];
+
+/**
+ * TODO: remove hardcoded names once suffixes are removed from Perfdocs
```
Comment (beatrice-acasandrei) [inline ui/perfherder/perf-helpers/constants.js:219]:
```suggestion
 * Used for building the documentation links (Perfdocs) and side-by-side links.
```

File: ui/perfherder/perf-helpers/helpers.js [hunk 5]
```diff
@@ -2,14 +2,15 @@ import moment from 'moment';
...
-import { getApiUrl } from '../../helpers/url';
+import { getApiUrl, getJobsUrl } from '../../helpers/url';
 import { update, processResponse } from '../../helpers/http';
 import PerfSeriesModel, {
   getSeriesName,
   getTestName,
 } from '../../models/perfSeries';
 import RepositoryModel from '../../models/repository';
 import JobModel from '../../models/job';
+import { sxsTaskName } from '../../helpers/constants';
```
Comment (beatrice-acasandrei) [inline ui/perfherder/perf-helpers/helpers.js:13]: `sxs` might not be obvious to everyone that it stands for side-by-side. Perhaps we could look into alternative names?

File: ui/perfherder/alerts/AlertTableRow.jsx (hunk 3 above)
Comment (beatrice-acasandrei) [inline ui/perfherder/alerts/AlertTableRow.jsx:61]: Try to move this logic to `render()`.

File: ui/perfherder/alerts/AlertTableRow.jsx [hunk 6]
```diff
@@ -45,14 +51,22 @@ export default class AlertTableRow extends React.Component {
...
+      alert.series_signature.framework_id === frameworks.browsertime &&
+      !alert.series_signature.tags.includes('interactive') &&
+      !browsertimeBenchmarksTests.includes(alert.series_signature.suite) &&
+      alert.backfill_record &&
+      !!alert.backfill_record.status;
 
     this.showCriticalMagnitudeIcons(alert);
+    this.setState({ showSideBySideLink });
```
Comment (beatrice-acasandrei) [inline ui/perfherder/alerts/AlertTableRow.jsx:69]: Check if it's really necessary to add `showSideBySideLink` to the state.

File: ui/perfherder/alerts/AlertTableRow.jsx (hunk 3 above)
Comment (esanuandra) [reply ui/perfherder/alerts/AlertTableRow.jsx:61]: I agree

File: ui/perfherder/alerts/AlertTable.jsx [hunk 7]
```diff
@@ -349,12 +355,21 @@ export default class AlertTable extends React.Component {
                         onChangeSort={this.onChangeSort}
                       />
                     </th>
-                    <th className="align-bottom">
+                    <th className="align-bottom text-nowrap">
```
Comment (esanuandra) [inline ui/perfherder/alerts/AlertTable.jsx:358]: `text-nowrap` is not needed on the Platform header. Better move it on Debug Tools header

File: ui/perfherder/alerts/AlertTable.jsx [hunk 8]
```diff
@@ -349,12 +355,21 @@ export default class AlertTable extends React.Component {
...
                       <TableColumnHeader
                         column={tableConfig.Platform}
                         onChangeSort={this.onChangeSort}
                       />
                     </th>
+                    {alertSummary.framework ===
+                      frameworkMapping.browsertime && (
+                      <th className="align-bottom">
```
Comment (esanuandra) [inline ui/perfherder/alerts/AlertTable.jsx:366]: Here we need the 'text-nowrap' class.

File: ui/perfherder/perf-helpers/helpers.js (hunk 5 above)
Comment (alexandru-io) [reply ui/perfherder/perf-helpers/helpers.js:13]: https://searchfox.org/mozilla-central/source/taskcluster/gecko_taskgraph/actions/side_by_side.py#65
https://github.com/search?q=repo%3Amozilla%2Ftreeherder%20sxs&type=code

Comment (alexandru-io) [reply ui/perfherder/perf-helpers/helpers.js:13]: @beatrice-acasandrei the name seemed not obvious to me too, but I'm not the one who chose it and there's a bunch of places on Treeherder and Mozilla central where it is used. The action task that triggers the side by side is called `gen-sxs`

File: ui/perfherder/alerts/AlertTableRow.jsx [hunk 9]
```diff
@@ -267,6 +273,29 @@ export default class AlertTableRow extends React.Component {
...
+    const { alert, alertSummary } = this.props;
+    const platform = alert.series_signature.machine_platform;
+    const { suite } = alert.series_signature;
+    let testName = suite;
+    if (suite in browsertimeEssentialTests) {
+      if ('bytecode-cached' in alert.series_signature.tags) {
+        testName = `bytecode ${suite}`;
+      } else {
+        testName = `essential ${suite}`;
+      }
```
Comment (esanuandra) [inline ui/perfherder/alerts/AlertTableRow.jsx:286]:
```suggestion
      testName = `essential ${suite}`;
      if ('bytecode-cached' in alert.series_signature.tags) {
        testName = `bytecode ${suite}`;
      }
```

File: ui/perfherder/alerts/AlertTable.jsx [hunk 10]
```diff
@@ -355,6 +361,14 @@ export default class AlertTable extends React.Component {
                         onChangeSort={this.onChangeSort}
                       />
                     </th>
+                    {alertSummary.framework === browsertime && (
+                      <th className="align-bottom text-nowrap">
+                        <TableColumnHeader
+                          column={tableConfig.DebuggingInformation}
+                          onChangeSort={this.onChangeSort}
```
Comment (beatrice-acasandrei) [inline ui/perfherder/alerts/AlertTable.jsx:368]: Sorting should be disabled, similar to the column `Tags & options`.

File: ui/perfherder/perf-helpers/constants.js [hunk 11]
```diff
@@ -204,3 +204,94 @@ export const noiseProfiles = {
...
+  'youtube-playback-v9-1080p60',
+  'youtube-playback-v9-full-1080p30',
+  'youtube-playback-v9-full-1080p60',
+  'youtube-playback-vp9-sfr',
+  'youtube-playback-widevine-h264-sfr',
+  'youtube-playback-widevine-hfr',
+  'youtube-playback-widevine-hfr',
+];
+
+export const browsertime = 13;
```
Comment (beatrice-acasandrei) [inline ui/perfherder/perf-helpers/constants.js:297]:
```suggestion
export const browsertimeId = 13;
```
Or a similar name that would suggest this constant is actually the id or pk for the framework browsertime.

File: ui/perfherder/alerts/AlertTableRow.jsx [hunk 12]
```diff
@@ -360,6 +389,16 @@ export default class AlertTableRow extends React.Component {
     const noiseProfileTooltip = alert.noise_profile
       ? noiseProfiles[alert.noise_profile.replace('/', '')]
       : noiseProfiles.NA;
+    const showSideBySideLink =
+      alert.series_signature.framework_id === browsertime &&
+      !alert.series_signature.tags.includes('interactive') &&
+      !browsertimeBenchmarksTests.includes(alert.series_signature.suite) &&
+      alert.backfill_record &&
+      (alert.backfill_record.status ===
```
Comment (beatrice-acasandrei) [inline ui/perfherder/alerts/AlertTableRow.jsx:397]: You could also use includes() for this check if you create an array for the statuses your interested in.
```suggestion
      (backfillResultStatuses.includes(alert.backfill_record.status)
```
Why are you interested in only these three statuses?

Comment (alexandru-io) [reply ui/perfherder/alerts/AlertTableRow.jsx:397]: I discovered just before I updated the PR that the statuses that are past the point the `side-by-side` jobs were triggered are just `backfilled`, `successful`, and `failed`.

File: ui/perfherder/perf-helpers/constants.js (hunk 11 above)
Comment (alexandru-io) [reply ui/perfherder/perf-helpers/constants.js:297]: the original intent was to have something like `frameworks.browsertime` for readability

File: ui/perfherder/alerts/AlertTableRow.jsx (hunk 12 above)
Comment (beatrice-acasandrei) [reply ui/perfherder/alerts/AlertTableRow.jsx:395]: @alexandru-io Could you also update the logic as mentioned above in the suggestion?

**Comment by esanuandra** [comment]
Looks good. We'll approve this after we have some tests to check the new `Debug tools` column.

**Review by beatrice-acasandrei — APPROVED** [review]
@alexandru-io Looks good, but if you plan to not add tests in this PR then please create a follow-up ticket for the test/s and link it in the description/comment here.

**Comment by alexandru-io** [comment]
> @alexandru-io Looks good, but if you plan to not add tests in this PR then please create a follow-up ticket for the test/s and link it in the description/comment here.

https://bugzilla.mozilla.org/show_bug.cgi?id=1841022

**Pull request merged**